Frameworks/QuartzCore/CATransaction.mm

```diff
@@ -192,72 +196,52 @@ + (CAAnimation*)_implicitAnimationForKey:(NSString*)forKey {
+    [self _currentLayerTransaction]->AddLayer([layer _layerProxy], nullptr, nullptr, nullptr);
 }
 
 + (void)_addSublayerToLayer:(CALayer*)layer sublayer:(CALayer*)sublayer {
-    GetCACompositor()->addNode([self _currentTransaction]->_transactionQueue,
-                               [sublayer _priv]->_presentationNode,
-                               [layer _priv]->_presentationNode,
-                               NULL,
-                               NULL);
+    [self _currentLayerTransaction]->AddLayer([sublayer _layerProxy], [layer _layerProxy], nullptr, nullptr);
 }
 
 + (void)_addSublayerToLayer:(CALayer*)layer sublayer:(CALayer*)sublayer before:(CALayer*)before {
-    GetCACompositor()->addNode([self _currentTransaction]->_transactionQueue,
-                               [sublayer _priv]->_presentationNode,
-                               [layer _priv]->_presentationNode,
-                               [before _priv]->_presentationNode,
-                               NULL);
+    [self _currentLayerTransaction]->AddLayer([sublayer _layerProxy], [layer _layerProxy], [before _layerProxy], nullptr);
 }
 
 + (void)_addSublayerToLayer:(CALayer*)layer sublayer:(CALayer*)sublayer after:(CALayer*)after {
-    GetCACompositor()->addNode([self _currentTransaction]->_transactionQueue,
-                               [sublayer _priv]->_presentationNode,
-                               [layer _priv]->_presentationNode,
-                               NULL,
-                               [after _priv]->_presentationNode);
+    [self _currentLayerTransaction]->AddLayer([sublayer _layerProxy], [layer _layerProxy], nullptr, [after _layerProxy]);
 }
 
 + (void)_replaceInLayer:(CALayer*)layer sublayer:(CALayer*)sublayer withSublayer:(CALayer*)newlayer {
-    GetCACompositor()->addNode([self _currentTransaction]->_transactionQueue,
-                               [newlayer _priv]->_presentationNode,
-                               [layer _priv]->_presentationNode,
-                               [sublayer _priv]->_presentationNode,
-                               NULL);
-    GetCACompositor()->removeNode([self _currentTransaction]->_transactionQueue, [sublayer _priv]->_presentationNode);
+    [self _currentLayerTransaction]->AddLayer([newlayer _layerProxy], [layer _layerProxy], [sublayer _layerProxy], nullptr);
+    [self _currentLayerTransaction]->RemoveLayer([sublayer _layerProxy]);
 }
 
 + (void)_moveLayer:(CALayer*)layer beforeLayer:(CALayer*)before afterLayer:(CALayer*)after {
-    GetCACompositor()->moveNode([self _currentTransaction]->_transactionQueue,
-                                [layer _priv]->_presentationNode,
-                                before ? [before _priv]->_presentationNode : NULL,
-                                after ? [after _priv]->_presentationNode : NULL);
+    [self _currentLayerTransaction]->MoveLayer([layer _layerProxy],
+                                               before ? [before _layerProxy] : nullptr,
```

Comment:
_layerProxy returns an std::shared_ptr<ILayerProxy>; not doing the nil check lead to AVs.

---
In reply to: [86900578](https://github.com/Microsoft/WinObjC/pull/1318#discussion_r86900578) [](ancestors = 86900578,86900544,86621323)